webpush/encrypt.go

```diff
@@ -270,7 +281,7 @@ func hkdf(salt, ikm, info []byte, length int) []byte {
 	mac = hmac.New(sha256.New, prk)
 	mac.Write(info)
 	mac.Write([]byte{1})
-	return mac.Sum(nil)[0:length]
+	return mac.Sum(nil)[0:length], nil
```

Comment:
Ok, to get to some compromise, between returning an error and panic, I would suggest panic since all callers of `hkdf` is internal and the `length` is not actually expected to be above 32 bits.

This will also eliminate `error` as the second return value introduced in this change and put back most of func signatures as they were before. 

Do you want to change this and then I'll review the rest of the code? It would be easier to see what actually chagned.